### PR TITLE
Improve some existing algorithms description

### PR DESCRIPTION
**Pull request by DelazJ**
Adding options for some gdal algorithms
Fixing typo and data type tag

### CODE REVIEW

File: source/docs/user_manual/processing_algs/gdal/vectormiscellaneous.rst [hunk 1]
```diff
@@ -207,6 +204,16 @@ Parameters
 
   Default: *False*
 
+``Promote to Multipart`` [boolean]
+  Casts features geometry type to multi in the output table.
```
Comment (yjacolin) [inline source/docs/user_manual/processing_algs/gdal/vectormiscellaneous.rst]: proposition: `multipart` intead of `multi`

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 2]
```diff
@@ -977,7 +979,7 @@ Parameters
   Default: *False*
 
 ``Output geometry has m dimension`` [boolean]
-  Choose if the output geometry should have the z dimension
+  Choose if the output geometry should have the m dimension
```
Comment (SrNetoChan) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: I have noticed there is no full stop here. Nonetheless, other places there are full stops in Parameters descriptions. (This is not a problem in this PR, just an observation)

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 3]
```diff
@@ -1791,7 +1793,7 @@ Parameters
   Optional
 
   X,Y coordinates of the point to rotate the features around.
-  If not set the rotation occurs around each feature's centroid. 
+  If not set the rotation occurs around each feature's centroid.
```
Comment (SrNetoChan) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Here's a full stop.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst (hunk 2 above)
Comment (DelazJ) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Good observation (and I can also observe that your last sentence does not have any) 😃 . It should have there. Unfortunately, this is inconsistent throughout the processing doc. I recall I mention this with Matteo  but while I can't remember what was the outcome, I think a full sentence should have a period at its end. Sometimes, we do not really have a sentence (in the sense subject--> verb --> complement)...

File: source/docs/user_manual/processing_algs/gdal/rasterextraction.rst [hunk 4]
```diff
@@ -115,6 +115,9 @@ Parameters
 ``Input layer`` [raster]
   Input elevation raster.
 
+``Band number`` [raster band]
+  Raster band to extract.
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/gdal/rasterextraction.rst]: @SrNetoChan eg should there be a full stop?

File: source/docs/user_manual/processing_algs/gdal/rasterextraction.rst [hunk 5]
```diff
@@ -129,12 +132,23 @@ Parameters
...
+``Produce 3D vector`` [boolean]
+
+  Default: *False*
+
+``Treat all raster values as valid`` [boolean]
+
+  Default: *False*
+
+``Input pixel value to treat as "nodata"`` [number]
+  Optional.
```
[the remark this answers is not in the document]
Comment (SrNetoChan) [reply source/docs/user_manual/processing_algs/gdal/rasterextraction.rst]: Not sure...

File: source/docs/user_manual/processing_algs/gdal/rasterextraction.rst (hunk 4 above)
Comment (SrNetoChan) [reply source/docs/user_manual/processing_algs/gdal/rasterextraction.rst]: I think it should.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst (hunk 2 above)
Comment (SrNetoChan) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: > Good observation (and I can also observe that your last sentence does not have any):-).

That doesn't count! :-P

**Comment by DelazJ** [comment]
@SrNetoChan I address the punctuation issue in another PR that needs this one merged beforehand. So if someone could give a look to the latest commits... Thanks.

**Pull request merged**